src/plugins/formulas/formulas.js

```diff
@@ -126,318 +166,559 @@ export class Formulas extends BasePlugin {
   }
 
   /**
-   * Returns cell value (evaluated from formula expression) at specified cell coords.
+   * Triggered on `updateSettings`.
    *
-   * @param {number} row Row index.
-   * @param {number} column Column index.
-   * @returns {*}
+   * @private
+   * @param {object} newSettings New set of settings passed to the `updateSettings` method.
    */
-  getCellValue(row, column) {
-    const cell = this.sheet.getCellAt(row, column);
+  updatePlugin(newSettings) {
+    this.engine.updateConfig(getEngineSettingsWithOverrides(this.hot.getSettings()));
+
+    const pluginSettings = this.hot.getSettings()[PLUGIN_KEY];
+
+    if (
+      isDefined(pluginSettings) &&
+      isDefined(pluginSettings.sheetName) &&
+      pluginSettings.sheetName !== this.sheetName
+    ) {
+      this.switchSheet(pluginSettings.sheetName);
+    }
+
+    // If no data was passed to the `updateSettings` method and no sheet is connected to the instance -> create a
+    // new sheet using the currently used data. Otherwise, it will be handled by the `afterLoadData` call.
+    if (!newSettings.data && this.sheetName === null) {
+      const sheetName = this.hot.getSettings()[PLUGIN_KEY].sheetName;
+
+      if (sheetName && this.engine.doesSheetExist(sheetName)) {
+        this.switchSheet(this.sheetName);
 
-    return cell ? (cell.getError() || cell.getValue()) : void 0;
+      } else {
+        this.sheetName = this.addSheet(sheetName ?? void 0, this.hot.getSourceDataArray());
+      }
+    }
+
+    super.updatePlugin(newSettings);
   }
 
   /**
-   * Checks if there are any formula evaluations made under specific cell coords.
-   *
-   * @param {number} row Row index.
-   * @param {number} column Column index.
-   * @returns {boolean}
+   * Destroys the plugin instance.
    */
-  hasComputedCellValue(row, column) {
-    return this.sheet.getCellAt(row, column) !== null;
+  destroy() {
+    unregisterEngine(this.engine, this.hot);
+
+    super.destroy();
   }
 
   /**
-   * Recalculates all formulas (an algorithm will choose the best method of calculation).
+   * Add a sheet to the shared HyperFormula instance.
+   *
+   * @param {string} [sheetName] The new sheet name. If not provided, will be auto-generated by HyperFormula.
```

Comment:
I've generated the docs based on this branch and this caught my eye.
![image](https://user-images.githubusercontent.com/571316/119482536-4526b800-bd54-11eb-9110-2bf095163d40.png)

I think that it looks weird when the non-last argument has the optional value. This is possible to do that `hot.getPlugin('formulas'). addSheet(undefined, [['data']])` but meh .. it's unnatural. Can we take the optional label off?

Reply:
`sheetData` should be optional as well. You may add sheet with `sheetName` but don't have to fill it with data

Reply:
I think they both should be optional, actually?

Reply:
@jansiegel so we allow calling without arguments `.addSheet()`? That's weird.

Reply:
Name does not have to be known by dev. HF will auto-generate the name `Sheet ${N}` and return `sheetId` for it

Reply:
It's basically an alias for https://handsontable.github.io/hyperformula/api/classes/crudoperations.html#addsheet

Reply:
Alias has a sense, it has only one optional argument. I'd remove the optional label and changed the docs that passing `null` or `undefined` (or one of them) will auto-generate the name. WDYT?

Reply:
Going further I think it is already working. It's only a matter of removing the square bracket from the docs.

Reply:
As I said, I'd go with both being optional https://github.com/handsontable/handsontable/pull/7648#discussion_r638665350.

Reply:
I was wrong :D with `hot.getPlugin('formulas').addSheet(null, [[1, 2, 3]])` I've got `Uncaught TypeError: Cannot read property 'toLowerCase' of null`. This works `hot.getPlugin('formulas').addSheet(undefined, [[1, 2, 3]])`.

Reply:
@budnix WDYT? ee3ecb5

Reply:
@jansiegel I'm happy with that 🏅